### PR TITLE
Add plum-dispatch

### PR DESCRIPTION
**Pull request by PhilipVinc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by wesselb** [comment]
I acknowledge I will be listed as a maintainer of this recipe.

File: recipes/plum-dispatch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+build:
+  number: 0
+  skip: true   # [py2k]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - setuptools >=41
```
Comment (xhochy) [inline recipes/plum-dispatch/meta.yaml:21]: Shouldn't this go into `host`?

Comment (PhilipVinc) [reply recipes/plum-dispatch/meta.yaml]: addressed

**Comment by PhilipVinc** [comment]
@xhochy I addressed all your comments and the tests are all passing.

File: recipes/plum-dispatch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  sha256: 766da66a93d0f00911777175dc7a43235ee3f18ead34c73977e710f732a918fd
+
+build:
+  number: 0
+  skip: true   # [py2k]
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (chrisburr) [inline recipes/plum-dispatch/meta.yaml:20]: Are you using a compiler?
```suggestion
```
If not this can likely be noarch.

Comment (PhilipVinc) [reply recipes/plum-dispatch/meta.yaml:20]: yes, this is compiling a Cython module

**Pull request merged**